plugins/fixed-edges/src/index.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/**
+ * @fileoverview A metrics manager that makes fixed edges more easily
+ * configurable.
+ */
+
+import Blockly from 'blockly/core';
+
+/**
+ * @typedef {{
+ *   top: (boolean|undefined),
+ *   bottom: (boolean|undefined),
+ *   left: (boolean|undefined),
+ *   right: (boolean|undefined)
+ * }}
+ */
+let FixedEdgesConfig;
+
+/**
+ * The current configuration for fixed edges.
+ * @type {!FixedEdgesConfig}
+ * @private
+ */
+const fixedEdges = {};
+
+/**
+ * The metrics manager for workspace metrics calculations with customizable
+ * fixed edges.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to calculate metrics
+ *     for.
+ * @implements {Blockly.IMetricsManager}
+ * @constructor
+ */
+export class FixedEdgesMetricsManager extends Blockly.MetricsManager {
+  /**
+   * Constructor for fixed edges workspace metrics manager.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
+   *     be added to.
+   */
+  constructor(workspace) {
+    super(workspace);
+  }
+
+  /**
+   * Sets which edges are fixed. This does not prevent fixed edges set by
+   * no scrollbars or single-direction scrollbars.
+   * @param {!FixedEdgesConfig} updatedFixedEdges The edges to set as fixed.
+   * @public
+   */
+  static setFixedEdges(updatedFixedEdges) {
+    fixedEdges.top = !!updatedFixedEdges.top;
+    fixedEdges.bottom = !!updatedFixedEdges.bottom;
+    fixedEdges.left = !!updatedFixedEdges.left;
+    fixedEdges.right = !!updatedFixedEdges.right;
+  }
+
+  /**
+   * Returns whether the scroll area has fixed edges.
+   * @return {boolean} Whether the scroll area has fixed edges.
+   * @package
+   * @override
+   */
+  hasFixedEdges() {
+    return true;
+  }
+
+  /**
+   * Computes the fixed edges of the scroll area.
+   * @param {!Blockly.MetricsManager.ContainerRegion=} opt_viewMetrics The view
+   *     metrics if they have been previously computed. Passing in null may
+   *     cause the view metrics to be computed again, if it is needed.
+   * @return {Blockly.MetricsManager.FixedEdges} The fixed edges of the scroll
```

Comment:
Update to `!Blockly.MetricsManager.FixedEdges`

Reply:
Updated. I'm also going to make a PR in core, since I shouldn't be changing the type for an overridden method.